src/go/rpk/pkg/cli/topic/produce.go

```diff
@@ -107,6 +116,18 @@ func newProduceCommand(fs afero.Fs, p *config.Params) *cobra.Command {
 				out.Die("invalid empty format")
 			}
 
+			isSchemaRegistry := schemaID > 0 || keySchemaID > 0
+			// If the user is using schema registry, we want to use the json
+			// format: %v{json} and %k{json}.
+			if isSchemaRegistry {
+				if strings.Contains(inFormat, "%v{") || strings.Contains(inFormat, "%k{") {
```

Comment:
nit: you could also use `strings.ContainsAny`, to simplify a bit more the conditional.